app/models/grda_warehouse/system_cohorts/currently_homeless.rb

```diff
@@ -0,0 +1,125 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module GrdaWarehouse::SystemCohorts
+  class CurrentlyHomeless < Base
+    def cohort_name
+      'Currently Homeless'
+    end
+
+    def sync
+      add_missing_clients
+
+      remove_housed_clients
+      remove_inactive_clients
+      remove_no_longer_meets_criteria
+    end
+
+    private def add_missing_clients
+      # Newly identified (first homeless enrollment in the past 2 years)
+      # for each client with ongoing enrollments not on cohort
+      #   find max exit or move in prior to min start of ongoing
+      # if no exit or move in within the 2 years before min start of ongoing - Newly identified
+      # if exit was to permanent destination within 2 year range, or move in was within 2 year range  - Returned from housing
+      # else Returned from inactive unless no service in INACTIVE period
+
+      moved_in_ph = enrollment_source.ongoing.ph.
+        where(she_t[:move_in_date].lt(Date.current)).
+        select(:client_id)
+
+      candidate_enrollments = homeless_enrollment_source.
+        entry.
+        ongoing.
+        where.not(client_id: service_history_source.where(date: Date.yesterday, homeless: false).select(:client_id)).
+        where.not(client_id: moved_in_ph).
+        where.not(client_id: cohort_clients.select(:client_id)).
+        group(:client_id).minimum(:first_date_in_program)
+
+      previous_enrollments = homeless_enrollment_source.
+        exit.
+        where(client_id: candidate_enrollments.keys).
+        order(greatest(she_t[:last_date_in_program], she_t[:move_in_date])).
+        pluck(:client_id, greatest(she_t[:last_date_in_program], she_t[:move_in_date]), :destination).
+        map { |client_id, *rest| [client_id, rest] }.to_h
+
+      most_recent_service_dates = service_history_source.
+        where(client_id: candidate_enrollments.keys).
+        group(:client_id).maximum(:date)
+
+      newly_identified = []
+      returned_from_housing = []
+      returned_from_inactive = []
+
+      candidate_enrollments.each do |client_id, enrollment_date|
+        previous_service_date, previous_destination = previous_enrollments[client_id]
+        if previous_service_date.blank? || previous_service_date < enrollment_date - 2.years
+          newly_identified << client_id
+        elsif HUD.permanent_destinations.include?(previous_destination) || previous_service_date < enrollment_date
+          returned_from_housing << client_id
+        elsif most_recent_service_dates[client_id] >= Date.current - days_of_inactivity.days
```

Comment:
I _think_ this should be `most_recent_service_dates[client_id] <= Date.current - days_of_inactivity.days`

Reply:
This is if they were declared inactive, and a new service item appears within the inactivity window, then make them active again.